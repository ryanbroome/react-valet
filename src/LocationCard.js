import React from "react";
import { Card, CardBody, CardTitle, CardSubtitle, ListGroup, ListGroupItem, CardText, Button } from "reactstrap";
// import { Link } from "react-router-dom";

const LocationCard = (location) => {
  return (
    <Card
      color="light"
      body
      className="text-center">
      <CardBody>
        <CardSubtitle>Sitename: {location.sitename}</CardSubtitle>
      </CardBody>
    </Card>
  );
};

export default LocationCard;
