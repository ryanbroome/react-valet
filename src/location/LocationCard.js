import React from "react";
import { Card, CardBody, CardTitle, CardSubtitle, ListGroup, ListGroupItem, CardText, Button } from "reactstrap";
// import { Link } from "react-router-dom";

const LocationCard = ({ location }) => {
  if (!location) {
    return null;
  }
  return (
    <Card
      color="light"
      body
      className="text-center">
      <CardBody>
        <CardTitle>Current Site: {location.sitename}</CardTitle>
        <CardSubtitle>More details to follow about location</CardSubtitle>
        <ListGroup>
          <ListGroupItem>Address to follow</ListGroupItem>
          <ListGroupItem>Hours to follow</ListGroupItem>
          <ListGroupItem>List of Active Valets to follow</ListGroupItem>
          <ListGroupItem>Total vehicles parked by location to follow</ListGroupItem>
        </ListGroup>
      </CardBody>
    </Card>
  );
};

export default LocationCard;
