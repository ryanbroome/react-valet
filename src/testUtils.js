import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
