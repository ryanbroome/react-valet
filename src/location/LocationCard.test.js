import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import LocationCard from "./LocationCard";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<LocationCard />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
  const location = { location: { sitename: "testSitename" } };
  render(
    <MemoryRouter>
      <MockUserContext children={<LocationCard location={{ location }} />}></MockUserContext>
    </MemoryRouter>
  );
});
