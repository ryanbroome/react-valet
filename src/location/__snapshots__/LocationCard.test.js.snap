// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center card card-body bg-light"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Current Site: 
      </h5>
      <h6
        class="card-subtitle"
      >
        More details to follow about location
      </h6>
      <ul
        class="list-group"
      >
        <li
          class="list-group-item"
        >
          Address to follow
        </li>
        <li
          class="list-group-item"
        >
          Hours to follow
        </li>
        <li
          class="list-group-item"
        >
          List of Active Valets to follow
        </li>
        <li
          class="list-group-item"
        >
          Total vehicles parked by location to follow
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
